tensorflow/core/grappler/optimizers/meta_optimizer.cc

```diff
@@ -165,10 +165,26 @@ bool MemoryOptimizerEnabled(
   return mem_opt_type != RewriterConfig::NO_MEM_OPT;
 }
 
+Status GetGraphDevice(const GraphDef& g_def, std::set<std::string>& devices) {
+  for (auto node : g_def.node()) {
```

Comment:
I think `auto` will do a copy here, you'll need `auto&`.

Reply:
Done